BindFamilyLockBonusActivationActivity/index.ts

```diff
@@ -0,0 +1,50 @@
+import { AzureFunction, Context } from "@azure/functions";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { BonusActivation } from "../generated/models/BonusActivation";
+import { CosmosDbDocumentCollection } from "../services/cosmosdb";
+import { generateFamilyUID } from "../utils/hash";
+
+const index: AzureFunction = async (_: Context, input: unknown) => {
+  const decoded = CosmosDbDocumentCollection.decode(input);
+  if (decoded.isLeft()) {
+    throw Error(
+      `BindFamilyLockBonusActivationActivity: cannot decode input [${readableReport(
+        decoded.value
+      )}]`
+    );
+  }
+  const documents = decoded.value.reduce(
+    (prev, rawDocument) => {
+      const errorOrBonusActivation = BonusActivation.decode(rawDocument);
+      // Skip invalid documents
+      if (errorOrBonusActivation.isLeft()) {
+        return prev;
+      }
+      const bonusActivation = errorOrBonusActivation.value;
+      return [
+        ...prev,
+        {
+          familyUID: generateFamilyUID(
```

Comment:
why do we generate the familyUID here ? isn't it inside the bonusActivation in input ?

Reply:
Maybe a bug, is missing inside the model type, but it is stored inside cosmos.

Reply:
try using 'RetrievedBonusActivation'

Reply:
Fixed in 4d6a562